### PR TITLE
More information for debugging.

### PR DESCRIPTION
**Pull request by zhangkun83**
- Include transport information in RPC's final status
- Implement toString() of transports

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/ClientCallImpl.java [hunk 1]
```diff
@@ -423,6 +423,14 @@ public final void runInContext() {
     @Override
     public void closed(Status status, Metadata trailers) {
       Long timeoutNanos = getRemainingTimeoutNanos(callOptions.getDeadlineNanoTime());
+      ClientTransport transport = null;
```
Comment (nmittler) [inline core/src/main/java/io/grpc/internal/ClientCallImpl.java]: Can this be done after we cancel?

Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/ClientCallImpl.java]: Done.

**Comment by nmittler** [comment]
LGTM

File: core/src/main/java/io/grpc/internal/ClientCallImpl.java [hunk 2]
```diff
@@ -434,7 +434,14 @@ public void closed(Status status, Metadata trailers) {
...
       }
-      final Status savedStatus = status;
+      ClientTransport transport = null;
+      try {
+        transport = transportFuture.get();
+      } catch (Exception e) {
+        // Ignore the exception, and keep transport as null.
+      }
+      final Status savedStatus = (transport == null
+          ? status : status.augmentDescription("transport=" + transport));
```
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/ClientCallImpl.java]: I don't think this is appropriate. A status description should be _exactly_ what was returned by the server (unless the server did not provide one).

Instead, we could maybe add another field to Status that is local-only.

**Pull request merged**